### PR TITLE
refactor(pkcs12): move to `expo/expo`

### PR DESCRIPTION
**Pull request by byCedric**
# Why

Moving the actively used `@expo/pkcs12` over from `expo/expo-cli`.

Currently used in:
- `eas-cli`
- Our backend

Counter-part PR: https://github.com/expo/expo-cli/pull/4794

# How

- Copied `@expo/pkcs12` from `expo/expo-cli`
- Updated `tsconfig.json` to point at this monorepo config
- Integrated `expo-module-scripts`
- Updated renamed types based on `@types/node@20`
  - `HexBase64Latin1Encoding` ([code](https://github.com/expo/expo/pull/25766/files#diff-56cd7ede5693f24e527da29965345ceb880ec8a8967bab3a993cafdfe677e44dR113)) → `BinaryToTextEncoding` ([types](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/68c795db12d031b4d16b13d6827571699571b9d4/types/node/crypto.d.ts#L390))
  - `Utf8AsciiLatin1Encoding` ([code](https://github.com/expo/expo/pull/25766/files#diff-56cd7ede5693f24e527da29965345ceb880ec8a8967bab3a993cafdfe677e44dR109)) → `Encoding` ([types](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/68c795db12d031b4d16b13d6827571699571b9d4/types/node/crypto.d.ts#L378))

# Test Plan

See if CI passes.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by EvanBacon — APPROVED** [review]
lgtm, but the codeowners isn't applicable since I didn't work on this library and don't know its purpose. Also unclear where we're using the package.

**Comment by byCedric** [comment]
> lgtm, but the codeowners isn't applicable since I didn't work on this library and don't know its purpose. Also unclear where we're using the package.

I'll change it, I think @quinlanj would be a good codeowner but will check before merging. It's being used in the EAS CLI and our backend, related to the [android keystores](https://github.com/expo/eas-cli/blob/b44ba6b22b2d0c1257c005fffa929f796044423c/packages/eas-cli/src/credentials/android/utils/keystoreNew.ts#L1).

File: .github/CODEOWNERS
Comment (ide) [inline .github/CODEOWNERS:13]: Could you please add Quin as a codeowner, too?

**Review by ide — APPROVED** [review]

File: .github/CODEOWNERS
Comment (byCedric) [reply .github/CODEOWNERS:13]:
```suggestion
/packages/@expo/pkcs12                          @bycedric @quinlanj
```

**Pull request merged**
